### PR TITLE
Add example of `instantiate` conversion strategies.

### LINKED ISSUE #2075: Update instantiate API doc (parameter conversion strategies section)
https://hydra.cc/docs/advanced/instantiate_objects/overview/#parameter-conversion-strategies

It's a bit hard to process this section, maybe we can provide some more examples here to illustrate the ideas better.

### PR DESCRIPTION
**Pull request by Jasha10**
Updating the docs on the `_convert_` argument to `instantiate`.
Closes #2075.

### CODE REVIEW

**Comment by Jasha10** [comment]
Here are the links to the website preview:
v1.1 docs: https://deploy-preview-2080--hydra-preview.netlify.app/docs/advanced/instantiate_objects/overview/#parameter-conversion-strategies
vNext docs: https://deploy-preview-2080--hydra-preview.netlify.app/docs/next/advanced/instantiate_objects/overview/#parameter-conversion-strategies

File: website/docs/advanced/instantiate_objects/overview.md [hunk 1]
```diff
@@ -206,14 +206,68 @@ OmegaConf containers (DictConfig, ListConfig).
 OmegaConf containers have many advantages over primitive dicts and lists but in some cases 
 it's desired to pass a real dicts and lists (for example, for performance reasons).
 
-You can change the parameter conversion strategy using the `_convert_` parameter (in your config or the call-site).
+You can change the parameter conversion strategy using the `_convert_` parameter.
```
Comment (jieru-hu) [inline website/docs/advanced/instantiate_objects/overview.md:209]: Thanks! looks great overall, a couple of nits:
1. it'd be great to provide more context on why DictConfig would be preferred some time.
2. it'd be great to provide a bit more details on the performance comments, ie, which option could potentially speed up the instantiate call

Comment (Jasha10) [reply website/docs/advanced/instantiate_objects/overview.md]: >     1. it'd be great to provide more context on why DictConfig would be preferred some time.

Done in https://github.com/facebookresearch/hydra/pull/2080/commits/dff94bf0a506e8d7184b86a72a6a951ab2bf5f1b.

>     2. it'd be great to provide a bit more details on the performance comments, ie, which option could potentially speed up the instantiate call

Done in https://github.com/facebookresearch/hydra/pull/2080/commits/f665b0759cefa1a8c11f5819f5b24d1af39f2f2f.

**Comment by Jasha10** [comment]
(Force-pushed to resolve merge conflict and incorporate recent CI fix.)

**Review by jieru-hu — APPROVED** [review]
sorry I missed this one. looks great, thank you!

**Pull request merged**